data/slash/common.go

```diff
@@ -19,38 +21,86 @@ type SlashingResult struct {
 type ProofTxData struct {
 	Round   uint64
 	ShardID uint32
+	ProofID ProofID
 }
 
-func getSortedHeadersV2(headersInfo []data.HeaderInfoHandler) (HeadersV2, error) {
-	if headersInfo == nil {
-		return HeadersV2{}, data.ErrNilHeaderInfoList
+type ProofID byte
+
+// Used by slashing notifier to create a slashing transaction
+// from a proof. Each transaction identifies a different
+// slashing event based on this ID
+const (
+	// MultipleProposalProofID = MultipleProposal's ID
+	MultipleProposalProofID ProofID = 0x1
+	// MultipleSigningProofID = MultipleSigning's ID
+	MultipleSigningProofID ProofID = 0x2
+)
+
+func sortAndGetHeadersV2(headersInfo []data.HeaderInfoHandler) (*HeadersV2, error) {
+	sortedHeaders, err := sortHeaders(headersInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	headersV2 := &HeadersV2{}
+	err = headersV2.SetHeaders(sortedHeaders)
+	if err != nil {
+		return nil, err
+	}
+
+	return headersV2, nil
+}
+
+func checkHeaderInfo(headerInfo data.HeaderInfoHandler, hashes map[string]struct{}) (string, error) {
```

Comment:
can you rename to getHeaderHash ?

Reply:
I renamed it to `getAndCheckUniqueHeaderHash` to better reflect its purpose